RELEASE_NOTES.rst

```diff
@@ -502,6 +502,12 @@ Airflow 2.3.0 restores the original meaning to the parameter. If you set it to "
 can be embedded in an iframe (no header is added), but when you set it to "false" the header is added
 and Airflow cannot be embedded in an iframe.
 
+``MySQLToS3Operator`` argument `pd_csv_kwargs` has beed replaced with `pd_kwargs` (#20807)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Although the old one (``pd_csv_kwargs``) is still accepted but no longer taken into account. Considering
+``MySQLToS3Operator`` deprecation it is recommeded to switch to ``SqlToS3Operator``. Doing this keep in mind
+that ``index`` and ``header`` arguments should be moved into ``pd_kwargs`` dict.
```

Comment:
This is not the right place.
The note needs to be in the provider change log not in the Airflow core release notes

Reply:
I think also amazon provider 5.0.0 removed it so only the new way will work.

If you believe that the change log of version 3.0.0 is not clear enough then you can add more information to it. PR #20807 was released in version 3.0.0

Reply:
Thank you for the quick response Elad!

I think it's important to list it here because this change can silently break data. In my case it happened—my custom line separator and column separator were ignored, because `pd_csv_kwargs` is not longer in use, and a lot of data became unusable which resulted in large amount of work.

Honestly speaking I'd consider it a bug. If it's a deprecation nothing should be broken but in this case it was.

Reply:
Sure yeah just add it to the correct file :)

Reply:
It was released in a major version (3.0.0) so if this is a breaking change you can just change the entry in the log from feature to breaking change

Reply:
I don't know yet how it is assumed to be but when I want to upgrade some tool I go and look at its change log or release notes. So when I was upgrading Airflow from 2.1.0 to 2.3.2 I looked into release notes, fixed all the things and still got a huge problem because of the mentioned change. It is not obvious that one should look into each provider change log.

How can I help to prevent things like this?

Reply:
What file is correct?

Reply:
Providers are independent from core. you need to look into provider change log as well.
You can upgrade providers regardless of upgrading airflow core
https://github.com/apache/airflow/blob/main/airflow/providers/amazon/CHANGELOG.rst

Reply:
Ok, I'll move the note where you suggest!

Will it be fine to mention somewhere that when upgrading Airflow it's important to take a look into provides' change log? It was not obvious in my case because I just switched from one official image (2.1.0) to another (2.3.2). There were no explicit providers update.

Reply:
It's explained in depth in https://github.com/apache/airflow#release-process-for-providers

Reply:
noting also that your assumption is incorrect.
https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html

You can install Airflow with providers and you can install providers as standalone. It's your choice if you want to couple it or not. Airflow core by itself does not bring provider dependencies.

Reply:
@jbmeerkat - but if you have an idea where to update it, you are absolutely free to do so. You are the best person in the world to make such PR- because a) you just had this problem b) now you know c) you know where you would look for it d) you can write it in the way that other users like you would search for it. 

So if you want to make a PR to add such a notice  you are most welcome. And it is super easy - just open the right page in the doc and click "Suggest a change on this page" and you will be able to edit it straight in the UI of GitHub. And you are most welcome to do it.

Reply:
This is an easy way to become one of more than 2100 contributors to Airflow. And a good way to pay back for the free software you use.

Reply:
It's eayssy to criticise somethign but in the open source if you see a problem, you contribute a fix rather than complain about it. And I encourage you to do so.